core/src/components/action-sheet/action-sheet.tsx

```diff
@@ -230,6 +230,18 @@ export class ActionSheet implements ComponentInterface, OverlayInterface {
    */
   @Method()
   async dismiss(data?: any, role?: string): Promise<boolean> {
+    /**
+     * When using an inline action sheet
+     * and presenting an action sheet it is possible to
+     * quickly dismiss the action sheet while it is
+     * presenting. We need to await any current
+     * transition to allow the present to finish
+     * before dismissing again.
+     */
+    if (this.currentTransition !== undefined) {
```

Comment:
Did we try to abstract this shared behavior across overlays to the [`dismiss`](https://github.com/ionic-team/ionic-framework/blob/main/core/src/utils/overlays.ts#L515) utility?

Since we are passing an instance of the class into the utility, we can access all the shared members we expect across implementations. 

Note: We likely would need to add `currentTransition` as a property to `OverlayInterface`.

Then, it might be possible to do this behavior once, in the utility, and all overlays would have the same consistent behavior. 

e.g.: Something such as:

```ts
if (overlay.currentTransition !== undefined) {
  await overlay.currentTransition;
}
// existing dismiss logic
```